src/azul/plugins/repository/tdr/__init__.py

```diff
@@ -221,6 +225,18 @@ def retrieve_rows(entity_type: str, entity_ids: Set[str]):
                 raise RuntimeError(f'Required entities not found in {self._source.bq_name}.{entity_type}: {entity_ids}')
         return bundle
 
+    def verify_authorization(self):
+        """
+        Verify that the current service account is authorized to read from the
+        TDR BigQuery tables.
+        """
+        try:
+            one(self._run_sql(f'SELECT * FROM {self._source.bq_name}.links LIMIT 1'))
+        except Forbidden:
+            tdr.on_auth_failure('the TDR BigQuery tables')
```

Comment:
The message is an implementation detail of on_auth_failure. Whether the string fits is brittle. Replace with boolean `bigquery` argument, True here, False below.